react/features/filmstrip/components/web/Thumbnail.js

```diff
@@ -0,0 +1,639 @@
+            _participant,
+            _participantCount,
+            isHovered
+        } = this.props;
+        const showConnectionIndicator = isHovered || !_connectionIndicatorAutoHideEnabled;
+        const { id, local = false, dominantSpeaker = false } = _participant;
+        const showDominantSpeaker = !_isDominantSpeakerDisabled && dominantSpeaker;
+        let statsPopoverPosition, tooltipPosition;
+
+        switch (_currentLayout) {
+        case LAYOUTS.TILE_VIEW:
+            statsPopoverPosition = 'right top';
+            tooltipPosition = 'right';
+            break;
+        case LAYOUTS.VERTICAL_FILMSTRIP_VIEW:
+            statsPopoverPosition = 'left top';
+            tooltipPosition = 'left';
+            break;
+        default:
+            statsPopoverPosition = 'top center';
+            tooltipPosition = 'top';
+        }
+
+        return (
+            <div>
+                <AtlasKitThemeProvider mode = 'dark'>
+                    { _connectionIndicatorDisabled
+                        ? null
+                        : <ConnectionIndicator
+                            alwaysVisible = { showConnectionIndicator }
+                            enableStatsDisplay = { true }
+                            iconSize = { iconSize }
+                            isLocalVideo = { local }
+                            participantId = { id }
+                            statsPopoverPosition = { statsPopoverPosition } />
+                    }
+                    <RaisedHandIndicator
+                        iconSize = { iconSize }
+                        participantId = { id }
+                        tooltipPosition = { tooltipPosition } />
+                    { showDominantSpeaker && _participantCount > 2
+                        ? <DominantSpeakerIndicator
+                            iconSize = { iconSize }
+                            tooltipPosition = { tooltipPosition } />
+                        : null }
+                </AtlasKitThemeProvider>
+            </div>);
+    }
+
+    /**
+     * Renders the avatar.
+     *
+     * @returns {ReactElement}
+     */
+    _renderAvatar() {
+        const { _participant } = this.props;
+        const { id } = _participant;
+        const styles = this._getStyles();
+
+        return (
+            <div
+                className = 'avatar-container'
+                style = { styles.avatarContainer }>
+                <Avatar
+                    className = 'userAvatar'
+                    participantId = { id } />
+            </div>
+        );
+    }
+
+    /**
+     * Renders the local participant's thumbnail.
+     *
+     * @returns {ReactElement}
+     */
+    _renderLocalParticipant() {
+        const {
+            _defaultLocalDisplayName,
+            _disableProfile,
+            _participant,
+            _videoTrack
+        } = this.props;
+        const { id } = _participant || {};
+        const { audioLevel = 0 } = this.state;
```

Comment:
I guess we can go without it but theoretically if we can receive `undefined` from the listener and I'm not sure how it will be handled...

If you ask me I'd leave the default value. Alternatively if you prefer I can remove it and make sure that the `AudioLevelIndicator` component handles it correctly. WDYT?

Reply:
Yeah, please let's have the last component in the chain, that is, AudioLevelIndicator handle it.

Reply:
Fixed!